### PR TITLE
Add custom certificate CA support for s3

### PR DESCRIPTION
**Pull request by clyang82**
Signed-off-by: clyang82 <chuyang@redhat.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

use prometheus common tls_config to connect with s3 object store so that we can pass the certificate in.

part of thanos-io/thanos#4820

### CODE REVIEW

File: pkg/objstore/s3/s3_test.go [hunk 1]
```diff
@@ -163,6 +163,63 @@ http_config:
...
+    cert_file: /certs/cert.crt
+    key_file: /certs/key.key
+    server_name: server
+    insecure_skip_verify: false
+  `)
+	cfg, err := parseConfig(input)
+	testutil.Ok(t, err)
+
+	if cfg.HTTPConfig.TLSConfig.CAFile != "/certs/ca.crt" {
+		t.Errorf("parsing of ca_file failed: got %v, expected %v",
```
Comment (GiedriusS) [inline pkg/objstore/s3/s3_test.go:181]: Let's use `testutil.Equals(t, "/certs/ca.crt", cfg.HTTPConfig.TLSConfig.CAFile)` and so on.

File: pkg/objstore/s3/s3_test.go [hunk 2]
```diff
@@ -163,6 +163,63 @@ http_config:
...
+  tls_config:
+    insecure_skip_verify: false
+  `)
+	cfg, err := parseConfig(input)
+	testutil.Ok(t, err)
+	transport, err := DefaultTransport(cfg)
+	testutil.Ok(t, err)
+
+	if !transport.TLSClientConfig.InsecureSkipVerify {
+		t.Errorf("parsing of insecure_skip_verify failed: got %v, expected %v", transport.TLSClientConfig.InsecureSkipVerify, true)
```
Comment (GiedriusS) [inline pkg/objstore/s3/s3_test.go:219]: Let's use testutil.Equals everywhere (:

**Review by GiedriusS — DISMISSED** [review]
Some suggestions on the test but otherwise LGTM

**Comment by clyang82** [comment]
@GiedriusS Thanks for your comments. Please take another look.

**Review by yeya24 — APPROVED** [review]
LGTM. Thanks for the contribution.

**Pull request merged**